perun-core/src/main/java/cz/metacentrum/perun/core/impl/MembersManagerImpl.java

```diff
@@ -739,6 +739,23 @@ public MemberGroupStatus getUnifiedMemberGroupStatus(PerunSession sess, User use
 			namedParams.addValue("statuses", statusCodes);
 		}
 
+		String groupQueryJoin = "";
+		String groupQueryWhere = "";
+		String groupStatusesQueryString = "";
+		if (query.getGroup() != null) {
+			groupQueryJoin = " LEFT JOIN groups_members ON members.id=groups_members.member_id ";
+			groupQueryWhere = " AND groups_members.group_id = (:groupId) ";
+			namedParams.addValue("groupId", query.getGroup().getId());
+
+			if (query.getGroupStatuses() != null && !query.getGroupStatuses().isEmpty()) {
+				groupStatusesQueryString = " AND groups_members.source_group_status in (:groupStatuses) ";
+				List<Integer> groupStatusCodes = query.getGroupStatuses().stream()
+					.map(MemberGroupStatus::getCode)
+					.collect(Collectors.toList());
+				namedParams.addValue("groupStatuses", groupStatusCodes);
```

Comment:
I have realized that the support for the group filtering will not be that easy. The group `group_members` table can contain multiple records for a single member because of indirect group memberships. We probably need to `merge` such records in the SQL and evaluate the overall member status. I think this might require vast changes to be made to the SQL.

An example of the problem. Suppose we have a member in groups A, A:B. In the table we would have something like this:
|member_id | group_id  |source_group_id|source_group_status|
--- | --- | --- | ---
|22|3|3|VALID|
|22|3|4|EXPIRED|

With the current SQL, this member would be returned if we asked for all expired members. However, he is not expired.